src/parser/grammar.y

```diff
@@ -244,27 +244,27 @@ edgeLength(A) ::= . {
 
 // *minHops..maxHops
 edgeLength(A) ::= MUL INTEGER(B) DOTDOT INTEGER(C). {
-	A = New_AST_LinkLength(B.intval, C.intval);
+	A = New_AST_LinkLength(B.intval, C.intval, true, true);
 }
 
 // *minHops..
 edgeLength(A) ::= MUL INTEGER(B) DOTDOT. {
-	A = New_AST_LinkLength(B.intval, UINT_MAX-1);
+	A = New_AST_LinkLength(B.intval, UINT_MAX-1, true, false);
 }
 
 // *..maxHops
 edgeLength(A) ::= MUL DOTDOT INTEGER(B). {
-	A = New_AST_LinkLength(1, B.intval);
+	A = New_AST_LinkLength(1, B.intval, false, true);
 }
 
 // *hops
 edgeLength(A) ::= MUL INTEGER(B). {
-	A = New_AST_LinkLength(1, B.intval);
+	A = New_AST_LinkLength(1, B.intval, false, true);
```

Comment:
Instead of this, I think we could get the job done with:
`A = New_AST_LinkLength(B.intval, B.intval);`

Reply:
You're right, I've misinterpreted the documentation.
I've added the logic which converts fixed relation length into an algebraic expression, as if the user had specified an explicit pattern ()-[]->()-[]->()... this way we enjoy the benefit of matrix multiplication.
For other variable length relations [*n..m] the traversal isn't performed by matrix multiplication, in this case we're using DFS and matrix are used as lookup tables.